R/spliceHelperFunctions.R

```diff
@@ -480,27 +484,39 @@ dropGrangesListElementsByWidth <- function(grangesList, minWidth=5, cutStartEnd=
 #' @param stand strand
 #' @noRd
 selectStartExonFromRangesList <- function(range, strand){
-  posStrandRng <- range[which(strand == "+")]
-  negStrandRng <- range[which(strand == "-")]
-  negStrandRng <- negStrandRng[which(end(negStrandRng) == max(end(negStrandRng)))]
-  startRng <- c(posStrandRng[which(start(posStrandRng) == min(start(posStrandRng)))],
-                negStrandRng[which(end(negStrandRng) == max(end(negStrandRng)))])
-  startRng <- startRng[sort(order(startRng)[names(range)])]
-  return (startRng)
+  partitioning<-PartitioningByEnd(cumsum(pmin(elementNROWS(range),1)), names=NULL)
+  largestExons <- as.numeric(cumsum(elementNROWS(range)))
+  startExonsSet <- c(1, largestExons[1:length(largestExons)-1]+1)
+  startExonsSet[strand == "-"] <- largestExons[strand == "-"]
+  return(relist(unlist(range, use.names = FALSE)[startExonsSet],partitioning))
 }
 
 #' Function that selects the last exon from an IRangesList object (exon_rank is required)
 #' @param range IRangesList
 #' @param stand strand
 #' @noRd
 selectEndExonFromRangesList <- function(range, strand){
-  posStrandRng <- range[which(strand == "+")]
-  negStrandRng <- range[which(strand == "-")]
-  endRng <- c(posStrandRng[which(end(posStrandRng) == max(end(posStrandRng)))], 
-              negStrandRng[which(start(negStrandRng) == min(start(negStrandRng)))])
-  endRng <- endRng[sort(order(endRng)[names(range)])]
-  return (endRng)
+  partitioning<-PartitioningByEnd(cumsum(pmin(elementNROWS(range),1)), names=NULL)
```

Comment:
same comments as above

Reply:
I have changed it in the version with relist().